app/controllers/results_controller.rb

```diff
@@ -336,13 +347,22 @@ def codeviewer
     @assignment = Assignment.find(params[:assignment_id])
     @submission_file_id = params[:submission_file_id]
     @focus_line = params[:focus_line]
-
+    @grouping = @current_user.grouping_for(Integer(params[:assignment_id]))
```

Comment:
These `Integer` conversions are almost never necessary, so you should check to see if there's another way of writing this code if you're encountering an error.